ddl/db_test.go

```diff
@@ -2639,3 +2639,33 @@ func (s *testDBSuite) TestAlterTableDropPartition(c *C) {
 	sql4 := "alter table table4 drop partition PAR0;"
 	s.testErrorCode(c, sql4, tmysql.ErrDropPartitionNonExistent)
 }
+
+func (s *testDBSuite) TestPartitionAddIndex(c *C) {
+	s.tk.MustExec("use test")
+	s.tk.MustExec("set @@tidb_enable_table_partition = 1")
+	s.tk.MustExec(`create table partition_add_idx (
+	id int not null,
+	hired date not null
+	)
+	partition by range( year(hired) ) (
+	partition p1 values less than (1991),
+	partition p3 values less than (2001),
+	partition p4 values less than (2004),
+	partition p5 values less than (2008),
+	partition p6 values less than (2012),
+	partition p7 values less than (2018)
+	);`)
+	for i := 0; i < 500; i++ {
+		s.tk.MustExec(fmt.Sprintf("insert into partition_add_idx values (%d, '%d-01-01')", i, 1988+rand.Intn(30)))
+	}
+
+	s.tk.MustExec("alter table partition_add_idx add index idx1 (hired)")
```

Comment:
Could we add a test like `testDBSuite.TestAddIndex`? We execute some DML while executing the operation of `add index`.

Reply:
Done.